src/llfs/ioring_log_device2.ipp

```diff
@@ -350,20 +383,37 @@ inline void IoRingLogDriver2<StorageT>::start_flush(CommitPos observed_commit_po
       }
     }
 
+    // If burst mode optimization is enabled, then round slot_range.upper_bound *down* to get the
+    // aligned upper bound (instead of up, the default); this way, we are less likely to need to
+    // issue another I/O to fill in partial data in flush_tail_.
+    //
+    if (this->flush_tail_ && this->options_.optimize_burst_mode) {
+      this->metrics_.burst_mode_checked.add(1);
+
+      const usize new_upper_bound = slot_max(
+          slot_range.lower_bound,
+          batt::round_down_bits(this->config_.data_alignment_log2, slot_range.upper_bound));
+
+      if (slot_range.upper_bound != new_upper_bound) {
+        this->metrics_.burst_mode_applied.add(1);
+      }
+
+      slot_range.upper_bound = new_upper_bound;
+    }
+
     // Align to 512-byte boundaries for direct I/O
     //
     SlotRange aligned_range = this->get_aligned_range(slot_range);
 
     // If this flush would overlap with an ongoing one (at the last device page) then trim the
-    // aligned_range so it doesn't.
+    // aligned_range (on the lower end) so it doesn't.
     //
-    if (this->flush_tail_) {
-      if (slot_less_than(aligned_range.lower_bound, this->flush_tail_->upper_bound)) {
-        aligned_range.lower_bound = this->flush_tail_->upper_bound;
-        if (aligned_range.empty()) {
-          flush_upper_bound = this->flush_tail_->upper_bound;
-          continue;
-        }
+    if (this->flush_tail_ &&
+        slot_less_than(aligned_range.lower_bound, this->flush_tail_->upper_bound)) {
+      aligned_range.lower_bound = this->flush_tail_->upper_bound;
+      if (aligned_range.empty()) {
+        flush_upper_bound = this->flush_tail_->upper_bound;
+        continue;
```

Comment:
This is what I was pointing to - "overlapping page transfers".

Reply:
Right... added metric.